make/hotspot/lib/CompileJvm.gmk

```diff
@@ -100,7 +100,7 @@ endif
 
 DISABLED_WARNINGS_xlc := tautological-compare shift-negative-value
 
-DISABLED_WARNINGS_microsoft := 4624 4244 4291 4146 4127
+DISABLED_WARNINGS_microsoft := 4624 4244 4291 4146 4127 4722
```

Comment:
It is annoying that we don't document what these warnings are. :(

Reply:
I've made the same comment in the past.  This time I decided to do something about it:
https://bugs.openjdk.org/browse/JDK-8303618
Document disabled C/C++ warnings on Windows

Reply:
Looks like something I can help with, will assign the enhancement to myself

Reply:
@dholmes-ora Dropping this here to help the review: Warning 4722 warns about when a destructor never returns which may cause memory leaks since allocated resources may not be released, or more accurately when control flow of the code never leaves a destructor. An extreme example of this would be a destructor of a class or struct immediately calling exit to terminate a program entirely when an object is destroyed